src/components/Onboarding/Screens/Apps/AgreementSettings/index.js

```diff
@@ -0,0 +1,263 @@
+          onFileUpdated={handleCovenantFileChange}
+          description={
+            <>
+              Drag and drop a document here or <TextFileUploader /> to upload
+              your community covenant. By uploading a file, you agree to Gardens
+              uploading this file to IPFS.
+            </>
+          }
+          label="Covenant"
+          previewLabel={
+            <div
+              css={`
+                margin-top: ${2 * GU}px;
+              `}
+            >
+              <Link
+                onClick={() => {
+                  setCovenantOpened(true)
+                }}
+              >
+                Preview Covenant
+              </Link>
+            </div>
+          }
+          required
+        />
+        <DurationFields
+          label={
+            <Fragment>
+              Settlement Period
+              <Help hint="What is Settlement Period?">
+                <strong>Settlement Period</strong> is the amount of time the
+                proposal's creator has to either accept a settlement or raise
+                the dispute to{' '}
+                <Link href="https://1hive.gitbook.io/celeste/">Celeste</Link>{' '}
+                after a proposal has been challenged.
+              </Help>
+            </Fragment>
+          }
+          duration={challengePeriod}
+          onUpdate={handleChallengePeriod}
+          required
+        />
+        <AmountField
+          label={
+            <Fragment>
+              Challenge Fee
+              <Help hint="What is Challenge Fee?">
+                <strong>Challenge Fee</strong> is the amount of collateral
+                tokens that will be locked every time an action is challenged.
+              </Help>
+            </Fragment>
+          }
+          value={challengeAmount}
+          onChange={handleChallengeAmount}
+          required
+          unitSymbol={config.tokens.symbol}
+          wide
+        />
+        <AmountField
+          label={
+            <Fragment>
+              Action Fee
+              <Help hint="What is Action Fee?">
+                <strong>The Action Fee</strong> is the amount of collateral
+                tokens that will be locked every time an action is submitted.
+              </Help>
+            </Fragment>
+          }
+          value={actionAmount}
+          onChange={handleActionAmount}
+          required
+          unitSymbol={config.tokens.symbol}
+          wide
+        />
+        <Info>We recommend sticking with the default duration.</Info>
+        {formError && (
+          <Info
+            mode="error"
+            css={`
+              margin-top: ${2 * GU}px;
+            `}
+          >
+            {formError}
+          </Info>
+        )}
+      </div>
+      <CovenantModal
+        open={covenantOpened}
+        file={covenantFile}
+        onClose={() => setCovenantOpened(false)}
+      />
+      <Navigation
+        backEnabled
+        nextEnabled={Boolean(
```

Comment:
I am not sure about the best UX here. We should decide between:
1. Having the button enabled and handle errors with `validateAgreementSettings`.
2. Or having it disable like we are currently doing.

My personal opinion is going with option 1 as it gives more feedback to the user. With the current solution, option 2, the `validateAgreementSettings` handler is irrelevant. So we can remove it.